lib/rouge/lexers/sparql.rb

```diff
@@ -41,16 +41,17 @@ def self.keywords
         rule %r('''), Str::Single, :string_single_literal
         rule %r('), Str::Single, :string_single
 
-        rule %r([$?]\w+), Name::Variable
-        rule %r((\w*:)(\w+)?) do |m|
+        rule %r([$?][[:word:]]+), Name::Variable
+        rule %r(([[:word:]-]*)(:)([[:word:]-]+)?) do |m|
           token Name::Namespace, m[1]
-          token Str::Symbol, m[2]
+          token Operator, m[3]
+          token Str::Symbol, m[4]
```

Comment:
I think we would want to update the indices.

```suggestion
          token Operator, m[2]
          token Str::Symbol, m[3]
```